includes/html/functions.inc.php

```diff
@@ -1138,9 +1138,26 @@ function get_arrays_with_application($device, $app_id, $app_name, $category = nu
     // app_name contains a separator character? consider it
     $offset = substr_count($app_name, $separator);
 
-    foreach (glob($pattern) as $rrd) {
-        $filename = basename($rrd, '.rrd');
+    if (Config::get('rrdcached')) {
+        $apparray = [];
+        $pattern = sprintf('%s-%s-%s', 'app', $app_name, $app_id);
+        // Build the rrdtool cmd to list rrdfiles for this device
+        $rrdcmd = sprintf('%s list /%s --daemon %s', Config::get('rrdtool', 'rrdtool'), $device['hostname'], Config::get('rrdcached'));
+        $rrd_files = shell_exec($rrdcmd);
```

Comment:
This should move into some Class like one of these https://github.com/librenms/librenms/tree/master/LibreNMS/Util

Reply:
Makes sense, i'll see if i can get something working.